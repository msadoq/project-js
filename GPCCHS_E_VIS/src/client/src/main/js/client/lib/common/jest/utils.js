import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { reduxForm } from 'redux-form';
import { Provider } from 'react-redux';

/**
 * @param obj
 * @param propsStub
 * @param stateStub
 */
export const shallowRenderSnapshotInReduxForm = (
  obj, propsStub = {}, stateStub = {}
) => {
  const Decorated = reduxForm({ form: 'testForm' })(obj);
  const store = createStore(state => state, stateStub);
  const tree = shallow(
    <Provider store={store}>
      <Decorated
        {...propsStub}
      />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
};

/**
 * @param Obj
 * @param propsStub
 */
export const shallowRenderSnapshot = (Obj, propsStub = {}) => {
  const tree = shallow(
    <Obj
      {...propsStub}
    />
  );
  expect(tree).toMatchSnapshot();
};
