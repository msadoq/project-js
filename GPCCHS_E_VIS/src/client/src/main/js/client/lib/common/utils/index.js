export applyDependencyToApi from './applyDependencyToApi';
export createLoggingSelector from './createLoggingSelector';
