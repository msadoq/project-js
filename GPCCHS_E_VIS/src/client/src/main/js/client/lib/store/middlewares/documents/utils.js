import _ from 'lodash/fp';
import parameters from 'common/configurationManager';
import { getWorkspaceFolder } from 'store/reducers/hsc';
import { EXTENSIONS } from 'constants';

export const isView = docType => /^.*View$/.test(docType);

const getFilter = docType => (
  { name: docType, extensions: [EXTENSIONS[docType]] }
);

const allViewsFilters = _.keys(EXTENSIONS).filter(isView).map(getFilter);
const allViewsExtensions = _.flatMap(_.get('extensions'), allViewsFilters);

export const getOpenExtensionsFilters = (docType) => {
  if (docType && !isView(docType)) {
    return [getFilter(docType)];
  }
  return [
    { name: 'Any VIMA views', extensions: allViewsExtensions },
    ...allViewsFilters,
  ];
};

export const getSaveExtensionsFilters = docType => [getFilter(docType)];

// warning, this selector is impure, do not memoize it
export const getDefaultFolder = state => (
  getWorkspaceFolder(state) || parameters.get('ISIS_DOCUMENTS_ROOT')
);
