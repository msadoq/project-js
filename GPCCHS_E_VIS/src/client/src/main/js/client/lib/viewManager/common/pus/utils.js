
import _ from 'lodash/fp';
import {
  VM_VIEW_PUS05,
  VM_VIEW_PUS11,
  VM_VIEW_PUS12,
  VM_VIEW_PUS13,
  VM_VIEW_PUS14,
  VM_VIEW_PUS15,
  VM_VIEW_PUS18,
  VM_VIEW_PUS19,
  VM_VIEW_PUS140,
  VM_VIEW_PUS142,
  VM_VIEW_PUS144,
  VM_VIEW_PUSMME,
} from 'viewManager/constants';
import { addTooltipWithContent } from './tooltip';
import parameters from '../../../common/configurationManager';
import constants from '../../../constants';


export const getModelEntryByDataType = (dataType) => {
  const deltaKeys = {
    [constants.Pus005OnBoardEvent]: constants.PUS005_ON_BOARD_EVENT,
    [constants.Pus005OnBoardEvent]: constants.PUS005_ON_BOARD_EVENT,
    [constants.Pus011CommandType]: constants.PUS011_COMMAND,
    [constants.Pus011SubScheduleType]: constants.PUS011_SUB_SCHEDULE,
    [constants.Pus011ApidType]: constants.PUS011_APID,
    [constants.Pus012ParameterMonitoringDefinitionType]:
      constants.PUS012_PARAMETER_MONITORING_DEFINITION,
    [constants.Pus013DownlinkLDTType]: constants.PUS013_LDT_PART,
    [constants.Pus013UplinkLDTType]: constants.PUS013_LDT_PART,
    [constants.Pus014ForwardedPacketType]: constants.PUS014_TM_PACKET,
    [constants.Pus015PacketStoreType]: constants.PUS015_PACKET_STORE,
    [constants.Pus018ObcpType]: constants.PUS018_OBCP,
    [constants.Pus019EventActionType]: constants.PUS019_EVENT_ACTION,
    [constants.Pus140ParameterType]: constants.PUS140_PARAMETER,
    [constants.Pus142FunctionalMonitoringType]: constants.PUS142_FUNCTIONAL_MONITORING,
    [constants.Pus144OnBoardFileType]: constants.PUS144_ON_BOARD_FILES,
    [constants.PusMmePacketType]: constants.PUSMME_PACKET,
  };

  return deltaKeys[dataType] || null;
};

export const getViewType = (pusService) => {
  const pusServices = {
    [constants.PUS_SERVICE_05]: VM_VIEW_PUS05,
    [constants.PUS_SERVICE_11]: VM_VIEW_PUS11,
    [constants.PUS_SERVICE_12]: VM_VIEW_PUS12,
    [constants.PUS_SERVICE_13]: VM_VIEW_PUS13,
    [constants.PUS_SERVICE_14]: VM_VIEW_PUS14,
    [constants.PUS_SERVICE_15]: VM_VIEW_PUS15,
    [constants.PUS_SERVICE_MME]: VM_VIEW_PUSMME,
    [constants.PUS_SERVICE_18]: VM_VIEW_PUS18,
    [constants.PUS_SERVICE_19]: VM_VIEW_PUS19,
    [constants.PUS_SERVICE_140]: VM_VIEW_PUS140,
    [constants.PUS_SERVICE_142]: VM_VIEW_PUS142,
    [constants.PUS_SERVICE_144]: VM_VIEW_PUS144,
  };
  return pusServices[pusService] || null;
};

export const getTableIdsByPusService = (pusService) => {
  const pusServices = {
    [constants.PUS_SERVICE_05]: [constants.PUS005_ON_BOARD_EVENT],
    [constants.PUS_SERVICE_11]: [
      constants.PUS011_APID,
      constants.PUS011_COMMAND,
      constants.PUS011_SUB_SCHEDULE,
    ],
    [constants.PUS_SERVICE_12]: [constants.PUS012_PARAMETER_MONITORING_DEFINITION],
    [constants.PUS_SERVICE_13]: [constants.PUS013_LDT_PART],
    [constants.PUS_SERVICE_14]: [constants.PUS014_TM_PACKET],
    [constants.PUS_SERVICE_15]: [constants.PUS015_PACKET_STORE],
    [constants.PUS_SERVICE_MME]: [constants.PUSMME_PACKET],
    [constants.PUS_SERVICE_18]: [constants.PUS018_OBCP],
    [constants.PUS_SERVICE_19]: [constants.PUS019_EVENT_ACTION],
    [constants.PUS_SERVICE_140]: [constants.PUS140_PARAMETER],
    [constants.PUS_SERVICE_142]: [constants.PUS142_FUNCTIONAL_MONITORING],
    [constants.PUS_SERVICE_144]: [constants.PUS144_ON_BOARD_FILES],
  };
  return pusServices[pusService] || null;
};

export const getViewServiceFromType = (type) => {
  const pusTypes = {
    [VM_VIEW_PUS05]: [constants.PUS_SERVICE_05],
    [VM_VIEW_PUS11]: [constants.PUS_SERVICE_11],
    [VM_VIEW_PUS12]: [constants.PUS_SERVICE_12],
    [VM_VIEW_PUS13]: [constants.PUS_SERVICE_13],
    [VM_VIEW_PUS14]: [constants.PUS_SERVICE_14],
    [VM_VIEW_PUS15]: [constants.PUS_SERVICE_15],
    [VM_VIEW_PUSMME]: [constants.PUS_SERVICE_MME],
    [VM_VIEW_PUS18]: [constants.PUS_SERVICE_18],
    [VM_VIEW_PUS19]: [constants.PUS_SERVICE_19],
    [VM_VIEW_PUS140]: [constants.PUS_SERVICE_140],
    [VM_VIEW_PUS142]: [constants.PUS_SERVICE_142],
    [VM_VIEW_PUS144]: [constants.PUS_SERVICE_144],
  };
  return pusTypes[type] || null;
};

/**
 * @param boolKey boolean condition to display key value
 * @param key displayed if boolKey, or empty
 * @param tooltip matching tooltip to be added
 * @returns { key [, tooltip]} new object with updated key and tooltip
 */
export const bindToBoolKey = ([boolKey, key, tooltip], store) => {
  const updateTypes = parameters.get('PUS_CONSTANTS').UPDATE_TYPE;
  let newStore = _.pick([key, tooltip], store);
  if (_.get(boolKey, store)) {
    newStore = _.set(
      tooltip,
      updateTypes[String(_.getOr(200, tooltip, newStore))],
      newStore
    );
  } else {
    newStore = _.set(key, '', newStore);
  }
  return newStore;
};

export const tableOverrideStyle = statusKeyList =>
  ({ content }) => {
    const statusColors = parameters.get('PUS_CONSTANTS').STATUS_COLOR;
    const { value, colKey } = content;

    // empty field style within StatusKeyList should not be overrided
    if (typeof value === 'string' && value.length === 0) {
      return {};
    }
    if (statusKeyList.indexOf(colKey) > -1) {
      return { backgroundColor: statusColors[value] };
    }
    return {};
  };

export const tableModifier = tooltips =>
  (cellContent = {}, content = {}) => {
    const { colKey, value } = cellContent;
    if (typeof value === 'string' && value.length === 0) {
      return cellContent;
    }
    const toolT = tooltips[colKey];
    if (toolT === undefined) {
      return cellContent;
    }
    return addTooltipWithContent(
      cellContent,
      content,
      {
        lastUpdateMode: { key: toolT.mode },
        lastUpdateTime: { key: toolT.time },
      }
    );
  };
