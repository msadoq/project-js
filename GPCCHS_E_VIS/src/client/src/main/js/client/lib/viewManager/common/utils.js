import _ from 'lodash/fp';
import { buildFormulaForAutocomplete } from './index';

/**
 * @param data
 * @param id
 * @param defaultTimelineId
 * @returns {{name: *, connectedData: {formula: string, fieldX: string, timeline: *}}}
 * @pure
 */
export const parseDragData = (data, id, defaultTimelineId) => {
  const getComObject = _.propOr('UNKNOWN_COM_OBJECT', 0);
  const formula =
    buildFormulaForAutocomplete(
      data.catalogName,
      data.item,
      getComObject(data.comObjects),
      data.comObjectFields
    );
  return {
    name: data.item,
    connectedData: {
      formula,
      catalog: data.catalogName,
      catalogItem: data.item,
      comObject: getComObject(data.comObjects),
      comObjectField: data.comObjectFields || 'convertedValue',
      fieldX: 'onboardDate',
      unit: 'V',
      domain: data.domain || '*',
      timeline: defaultTimelineId,
    },
  };
};

export const dummy = (x, y) => x + y;
