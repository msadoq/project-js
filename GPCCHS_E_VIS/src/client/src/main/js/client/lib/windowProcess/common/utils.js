/* Here is some react utils */

const getWrappedInstance = i => (i.getWrappedInstance ? i.getWrappedInstance() : i);

export default {
  getWrappedInstance,
};
