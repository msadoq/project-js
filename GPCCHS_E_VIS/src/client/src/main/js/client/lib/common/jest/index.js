// ====================================================================
// HISTORY
// VERSION : 1.1.2 : FA : #6670 : 16/06/2017 : Move and rename jest.js in jest/setup.js + test.js in jest/index.js
// VERSION : 1.1.2 : FA : #6798 : 22/06/2017 : Multiple changes on the load mechansim of adapters : - To test with Jest, add a mock of config(MESSAGES_NAMESPACE) in jest/index.js - Test fix - Lint pass ( 1 test is still KO)
// VERSION : 1.1.2 : DM : #6700 : 28/06/2017 : Automatically freeze given state to mockStore (in common/jest)
// VERSION : 1.1.2 : FA : #7453 : 07/08/2017 : Change adapters path in tests
// END-HISTORY
// ====================================================================

import { tmpdir } from 'os';
import _ from 'lodash/fp';
import { resolve } from 'path';
import thunk from 'redux-thunk';
import deepFreeze from 'deep-freeze';
import configureMockStore from 'redux-mock-store';
import { loadStubs, getStubData } from '../../utils/stubs';
import flattenDataId from '../flattenDataId';
import { registerGlobal } from '../../utils/adapters';

const mockedAdaptersPath = [
  {
    path: './adapters/',
    ns: 'isis',
  },
  {
    path: './adapters/',
    ns: 'dc',
  },
];

const mockLoadStubs = () => loadStubs(mockedAdaptersPath);
const mockRegister = () => registerGlobal(mockedAdaptersPath);


const dataStub = getStubData();

const freezeMe = o => o && deepFreeze(o);

const freezeArgs = f => (...args) => {
  const frozenArgs = args.map(arg => freezeMe(arg));
  return f(...frozenArgs);
};

const mockStore = _.compose(configureMockStore([thunk]), freezeMe);

const testMemoization = (selector, state, ownProps) => {
  const newState = _.cloneDeep(state);
  const newOwnProps = _.cloneDeep(ownProps);
  selector.resetRecomputations();
  expect(selector.recomputations()).toBe(0);

  const result1 = selector(newState, newOwnProps);
  expect(selector.recomputations()).toBe(1);

  const result2 = selector(newState, newOwnProps);
  expect(selector.recomputations()).toBe(1);
  expect(result1).toEqual(result2);
};

const getRemoteId = override => flattenDataId(dataStub.getDataId(override));

const mockIpcReply = () => {
  //
};

module.exports = {
  mockStore, // thunk testing
  mockIpcReply, // IPC controller testing
  freezeMe, // reducers testing
  freezeArgs, // reducers testing
  testMemoization, // reselect testing
  mockRegister, // protobuf testing
  mockLoadStubs,
  getTmpPath: (...args) => resolve(tmpdir(), 'vima-tests', ...args), // documentManager testing
  getRemoteId,
};
