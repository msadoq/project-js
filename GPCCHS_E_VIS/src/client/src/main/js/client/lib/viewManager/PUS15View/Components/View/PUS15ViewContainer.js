import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import { connect } from 'react-redux';
import { PUS_SERVICE_15 } from 'constants';
import PUS15View from './PUS15View';
import { getWindowIdByViewId } from '../../../../store/selectors/windows';
import { getPUSViewData } from '../../../common/pus/dataSelectors';
import parameters from '../../../../common/configurationManager';
import { injectTabularData } from '../../../commonData/reducer';
import { bindToBoolKey } from '../../../common/pus/utils';

const pusService = PUS_SERVICE_15;

const mapStateToProps = (state, { viewId }) => {
  let data = getPUSViewData(state, { viewId, pusService });
  const statuses = parameters.get('PUS_CONSTANTS').STATUS;
  const updateTypes = parameters.get('PUS_CONSTANTS').UPDATE_TYPE;

  if (typeof data === 'object' && Object.keys(data).length > 0) {
    for (let i = 0; i < data.headers.length; i += 1) {
      data.headers[i].status = statuses[_.getOr(200, 'status', data.headers[i])];
      data.headers[i].serviceApid = _.getOr(null, 'serviceApid', data.headers[i]);
      data.headers[i].serviceApidName = _.getOr(null, 'serviceApidName', data.headers[i]);
      data.headers[i].uniqueId = _.getOr(null, 'uniqueId', data.headers[i]);
    }

    // injectTabularData: add data tables to dedicated injectTabularData (VirtualizedTableView)
    data = injectTabularData(
      data,
      'onBoardStorages',
      _.getOr([], ['dataForTables', 'pus015PacketStore'], data)
        .map(store => ({
          ..._.omit(['pus015Packet'], store),
          dumpEnabled: String(_.getOr('boolean', 'dumpEnabled', store)),
          downlinkStatus: statuses[String(_.getOr(200, 'downlinkStatus', store))],
          storeStatus: statuses[String(_.getOr(200, 'storeStatus', store))],
          lastUpdateModeStoreId: updateTypes[String(_.getOr(200, 'lastUpdateModeStoreId', store))],
          lastUpdateModeStoreType: updateTypes[String(_.getOr(200, 'lastUpdateModeStoreType', store))],
          lastUpdateModeStoreStatus: updateTypes[String(_.getOr(200, 'lastUpdateModeStoreStatus', store))],
          lastUpdateModeDownlinkStatus: updateTypes[String(_.getOr(200, 'lastUpdateModeDownlinkStatus', store))],
        }))
    );

    data = injectTabularData(
      data,
      'storageDef',
      _.getOr([], ['dataForTables', 'pus015PacketStore'], data)
        .reduce((acc, store) => [...acc, ...store.pus015Packet], [])
        .map(packet => ({
          ...packet,
          ...bindToBoolKey(['isSubsamplingRatioSet', 'subsamplingRatio', 'lastUpdateModeSubSamplingRatio'], packet),
          lastUpdateModePacketId: updateTypes[String(_.getOr(200, 'lastUpdateModePacketId', packet))],
        }))
    );

    data = _.omit(['dataForTables'], data);
  }

  const windowId = getWindowIdByViewId(state, { viewId });

  return {
    data,
    windowId,
  };
};

const PUS15ViewContainer = connect(mapStateToProps, null)(PUS15View);

PUS15ViewContainer.propTypes = { viewId: PropTypes.string.isRequired };

export default PUS15ViewContainer;
