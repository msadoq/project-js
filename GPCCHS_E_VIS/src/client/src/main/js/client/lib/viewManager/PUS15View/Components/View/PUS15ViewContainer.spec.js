import { shallowRenderSnapshotInReduxForm } from 'common/jest/utils';
import PUS15ViewContainer from './PUS15ViewContainer';

const stubState = {};

const propsStub = {
  serviceApid: 'serviceApid',
  serviceApidName: 'serviceApidName',
};

describe('viewManager/PUS15View/Components/View/PUS15ViewContainer', () => {
  test('snapshot', () => {
    shallowRenderSnapshotInReduxForm(PUS15ViewContainer, propsStub, stubState);
  });
});
