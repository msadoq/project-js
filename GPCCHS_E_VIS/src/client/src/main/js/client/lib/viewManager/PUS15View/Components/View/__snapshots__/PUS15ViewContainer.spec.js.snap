// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewManager/PUS15View/Components/View/PUS15ViewContainer snapshot 1`] = `
<ReduxForm
  serviceApid="serviceApid"
  serviceApidName="serviceApidName"
/>
`;
